source/tutorials/db-coll.txt

```diff
@@ -0,0 +1,279 @@
+
+Capped Collection
+~~~~~~~~~~~~~~~~~
+
+The following operation creates a capped collection limited to 1
+megabyte:
+
+.. code-block:: scala
+
+   database.createCollection("cappedCollection", CreateCollectionOptions().capped(true).sizeInBytes(0x100000))
+           .printResults()
+
+To learn more about capped collections, see :manual:`Capped Collections
+</core/capped-collections/>` in the Server manual.
+
+Document Validation
+~~~~~~~~~~~~~~~~~~~
+
+MongoDB allows you to validate documents during
+updates and inserts. Validation rules are specified on a collection
+level by using the ``ValidationOptions`` class, which takes a
+filter document that specifies the validation rules or expressions.
+
+The following example creates a collection with schema validation:
+
+.. code-block:: scala
+
+   ValidationOptions collOptions = ValidationOptions().validator(
+           Filters.or(Filters.exists("email"), Filters.exists("phone")))
+   
+   database.createCollection("contacts", CreateCollectionOptions().validationOptions(collOptions))
+           .printResults()
+
+To learn more about document validation, see :manual:`Schema Validation
+</core/schema-validation/>` in the Server manual.
+
+Get A List of Collections
+-------------------------
+
+You can get a list of the collections in a database by using the
+``MongoDatabase.listCollectionNames()`` method:
+
+.. code-block:: scala
+
+   database.listCollectionNames().printResults()
+
+Drop a Collection
+-----------------
+
+You can drop a collection and delete all of the data in the collection
+by using the ``MongoCollection.drop()`` method:
+
+.. code-block:: scala
+
+   val collection: MongoCollection[Document] = database.getCollection("contacts")
+   collection.drop().printResults()
+
+.. _scala-immutability:
+
+Immutability
+------------
+
+``MongoDatabase`` and ``MongoCollection`` instances are immutable. To
+create new instances from existing instances that have different
+properties, such as different :manual:`read concerns
+</reference/read-concern/>`, :manual:`read preferences
+</reference/read-preference/>`, and :manual:`write concerns
+</reference/write-concern/>`, the ``MongoDatabase`` and
+``MongoCollection`` class provides the following methods:
+
+- ``MongoDatabase.withReadConcern()``
+- ``MongoDatabase.withReadPreference()``
+- ``MongoDatabase.withWriteConcern()``
+- ``MongoCollection.withReadConcern()``
+- ``MongoCollection.withReadPreference()``
+- ``MongoCollection.withWriteConcern()``
+
+.. TODO To learn more, see the :ref:`scala-read-operations` and
+.. :ref:`scala-write-ops` tutorials.
+
+CodecRegistry
+-------------
+
+An overload of the ``getCollection()`` method allows you to specify a
+different class for representing BSON documents. For example, you
+might want to use the strict and type-safe ``BsonDocument`` class to
+model your documents when performing CRUD operations:
+
+.. code-block:: scala
+
+   // pass BsonDocument.class as the second argument
```

Comment:
Question:
Did this comment belong to a larger code block or different example? I think it should be removed if it isn't relevant.

Reply:
I think it might be an error in the source text. removed